tests/validation/tests/v3_api/test_workload.py

```diff
@@ -1004,65 +990,59 @@ def test_rbac_wl_daemonset_update(role, remove_resource):
+@pytest.mark.parametrize("config", WORKLOAD_TYPES)
+def test_rbac_wl_parametrize_update_negative(role, remove_resource, config):
     if role == CLUSTER_OWNER:
-        # cluster owner can edit daemonsets in any project in the cluster
-        project = rbac_get_project()
+        # workloads of type job cannot be edited
+        if config == "job":
+            return
+        # cluster owner can edit workloads in any project in the cluster
         user_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
-        p2 = rbac_get_unshared_project()
-        p_client = get_project_client_for_token(project, user_token)
-        p_client2 = get_project_client_for_token(p2, user_token)
-        ns2 = rbac_get_unshared_ns()
+        p_client, project, ns = setup_wl_project_by_role(role)
+        p_client2 = get_project_client_for_token(project, user_token)
         name = random_test_name("default")
         con = [{"name": "test1", "image": TEST_IMAGE}]
-        workload = p_client2.create_workload(name=name,
-                                             containers=con,
-                                             namespaceId=ns2.id,
-                                             daemonSetConfig={})
+        workload = create_workload_by_type(p_client2, name, con, ns, config)
         wait_for_wl_to_active(p_client2, workload)
         con = [{"name": "test1", "image": "nginx"}]
         p_client.update(workload, containers=con)
         remove_resource(workload)
     else:
-        project = rbac_get_unshared_project()
-        # roles cannot edit daemonsets in projects they cannot access
+        project2 = rbac_get_unshared_project()
+        user_token = rbac_get_user_token_by_role(role)
+        # roles cannot edit workloads in projects they cannot access
         # deploy a workload as cluster owner
         cluster_owner_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
         cluster_owner_p_client = get_project_client_for_token(
-                                project, cluster_owner_token)
+                                project2, cluster_owner_token)
         ns = rbac_get_namespace()
         con = [{"name": "test1", "image": TEST_IMAGE}]
         name = random_test_name("default")
-        workload = cluster_owner_p_client.create_workload(name=name,
-                                                          containers=con,
-                                                          namespaceId=ns.id,
-                                                          daemonSetConfig={})
+        workload = create_workload_by_type(cluster_owner_p_client,
+                                           name, con, ns, config)
         # project read-only can NOT edit existing workload
         with pytest.raises(ApiError) as e:
-            user_token = rbac_get_user_token_by_role(role)
-            project2, ns = create_project_and_ns(user_token, namespace["cluster"],
+            project3, ns = create_project_and_ns(user_token, namespace["cluster"],
```

Comment:
why do we create a new project here? Should we user the unshared project (project2)?

Reply:
Changed to use the unshared project. Addressed in changes.